### PR TITLE
DTSPB-2907 HMRC Data Extract

### PR DESCRIPTION
**Pull request by FeliTam**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPB-2907

### Change description ###
Add output fields to HMRC data extract

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/probate/service/filebuilder/HmrcFileService.java [hunk 1]
```diff
@@ -103,6 +103,14 @@ private int prepareCaseData(ImmutableList.Builder<String> fileData, Long id, Cas
             fileData.add(getPoundValue(data.getIhtNetValue()));
             fileData.add(DataExtractGrantType.valueOf(data.getCaseType()).getCaseTypeMapped());
             fileData.add(FINAL_GRANT);
+            fileData.add(getPoundValue(data.getIhtEstateGrossValue()));
+            fileData.add(getPoundValue(data.getIhtEstateNetValue()));
+            fileData.add(getPoundValue(data.getIhtEstateNetQualifyingValue()));
+            fileData.add(Optional.ofNullable(data.getIhtUnusedAllowanceClaimed()).orElse(""));
```
Comment (sanjayparekh-hmcts) [inline src/main/java/uk/gov/hmcts/probate/service/filebuilder/HmrcFileService.java]: maybe its worth putting this Optional.. instantiation into its own method, so you can reuse it?

File: src/test/java/uk/gov/hmcts/probate/service/filebuilder/HmrcFileServiceTest.java [hunk 2]
```diff
@@ -168,7 +184,15 @@ public void setup() {
             .registryLocation("Liverpool")
             .grantIssuedDate(grantIssuedDate)
             .solsSOTName("John The personal")
-            .applicationType(ApplicationType.PERSONAL);
+            .applicationType(ApplicationType.PERSONAL)
```
Comment (sanjayparekh-hmcts) [inline src/test/java/uk/gov/hmcts/probate/service/filebuilder/HmrcFileServiceTest.java:187]: do you need unit tests for when some of these estate values are null?

**Review by sanjayparekh-hmcts — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/probate/service/filebuilder/HmrcFileServiceTest.java (hunk 2 above)
Comment (FeliTam) [reply src/test/java/uk/gov/hmcts/probate/service/filebuilder/HmrcFileServiceTest.java:187]: getPoundValue has handled null values already.
Tests using caseDataMissingData setup have covered null values for new added fields.

Comment (sanjayparekh-hmcts) [reply src/test/java/uk/gov/hmcts/probate/service/filebuilder/HmrcFileServiceTest.java:187]: what about the Optional setup on lines109 onwards - put that into a method?

**Review by sanjayparekh-hmcts — APPROVED** [review]
